File: src/python/receive.py

import zmq
import msgpack
import sys

if len(sys.argv) < 3:
	print "usage: ", sys.argv[0], " bind|connect port"
	exit(1)

context = zmq.Context()

receiver = context.socket(zmq.PULL)
if sys.argv[1] == "bind":
	receiver.bind(sys.argv[2])
elif sys.argv[1] == "connect":
	receiver.connect(sys.argv[2])
else:
	print "bind or connect"
	exit(1)
	
while 1:
	message = receiver.recv()
	print msgpack.unpackb(message, encoding='utf-8')
	
receiver.close()
context.term()

exit(0)

File: src/javascript/receive.js
var zmq = require('/usr/local/lib/node_modules/zmq'), 
	msgpack = require('/usr/local/lib/node_modules/msgpack');

if (process.argv.length < 4) {
	console.log("usage: ", process.argv[0], process.argv[1], " bind|connect port");
	process.exit(1);
}

var receiver = zmq.socket('pull');

if (process.argv[2] == "bind") {
	receiver.bind(process.argv[3]);
}
else if (process.argv[2] == "connect") {
	receiver.connect(process.argv[3]);
}
else {
	console.log("bind or connect");
	process.exit(1);
}

receiver.on('message', function(message) {
	console.log(msgpack.unpack(message));
});
